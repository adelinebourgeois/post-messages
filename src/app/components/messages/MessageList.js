import React, {useEffect} from "react";
import PropTypes from 'prop-types';
import {useDispatch, useSelector} from "react-redux";

import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import Divider from "@material-ui/core/Divider";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";


const useStyles = makeStyles((theme) => ({
    root: {
        maxWidth: '100%',
    },
    inline: {
        display: 'inline',
    },
    text: {
        padding: theme.spacing(2, 2, 0),
    },
    paper: {
        paddingBottom: 50,
    },
}));

const MessageList = () => {
    // const dispatch = useDispatch; Permet d'envoyer une action au reducer
    const messages = useSelector(state => state.messages);
    const classes = useStyles();

    // Call Ã  l'api pour fetch tout les messages

    // useEffect(() => {
    //      fetchMessages();
    // }, [])

    // const fetchMessages = async () => {
    //      try {
    //          const response = await fetch("https://api.example.com/messages");
    //          const data = await response.json();
    //          dispatch({type:'initial', payload: data});
    //      } catch (e) {
    //          dispatch({type: 'error', payload: e});
    //      }
    // }

    return (
        <>
            <Typography className={classes.text} variant="h5" gutterBottom>
                Messages
            </Typography>
            <List className={classes.root}>
                {messages.map(message => {
                    return (
                        <>
                            <ListItem alignItems="flex-start" key={message.id}>
                                <ListItemAvatar>
                                    <Avatar alt={message.userName} src="/static/images/avatar/1.jpg" />
                                </ListItemAvatar>
                                <ListItemText
                                    primary={<React.Fragment>
                                        <Typography
                                            component="span"
                                            variant="body2"
                                            className={classes.inline}
                                            color="textPrimary"
                                        >
                                            {message.text}
                                        </Typography>
                                    </React.Fragment>}
                                    secondary={message.type}
                                />
                            </ListItem>
                            <Divider variant="fullWidth" component="li" />
                        </>
                    )
                })}
            </List>
        </>
    )

}

MessageList.defaultProps = {
    messages: [],
}

MessageList.propTypes = {
    messages: PropTypes.array,
};

export default MessageList;