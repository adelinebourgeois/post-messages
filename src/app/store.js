import { createStore } from 'redux';
import messagesReducer from "./reducers/messages";

const store = createStore(messagesReducer)

export default store
