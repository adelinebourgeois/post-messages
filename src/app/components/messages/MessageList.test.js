import React from "react";
import * as reactRedux from 'react-redux';
import {render} from '@testing-library/react'
import {beforeEach, describe, expect, it} from "@jest/globals";
import MessageList from "./MessageList";
import store from './../../store';

describe("MessageList", () => {
    // mocking useSelector on our mock store
    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

    // mocks are cleared before each test case
    beforeEach(() => {
        useSelectorMock.mockClear()
    })

    it('should render ListItem if messages.length > 0', () => {
        useSelectorMock.mockReturnValue(store.getState().messages);
        const component = render(<MessageList />);

        expect(component).toMatchSnapshot();
    })
})
