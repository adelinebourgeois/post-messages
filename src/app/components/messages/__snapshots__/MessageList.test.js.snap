// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render ListItem if messages.length > 0 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h5
        class="MuiTypography-root makeStyles-text-3 MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Messages
      </h5>
      <ul
        class="MuiList-root makeStyles-root-1 MuiList-padding"
      >
        <div>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Lara Croft"
                  class="MuiAvatar-img"
                  src="/static/images/avatar/1.jpg"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <span
                  class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
                >
                  Vivamus hendrerit ipsum eros, non vulputate elit cursus sed. Proin fringilla vestibulum lobortis. Vivamus blandit diam vitae massa iaculis blandit. Sed molestie pharetra nisl, sed lobortis nunc venenatis in. Duis vel leo id elit vestibulum iaculis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent felis augue, aliquam quis nisl et, sollicitudin tincidunt augue.
                </span>
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                public
              </p>
            </div>
          </li>
          <li
            class="MuiDivider-root"
            role="separator"
          />
        </div>
        <div>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Nathan Drake"
                  class="MuiAvatar-img"
                  src="/static/images/avatar/1.jpg"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <span
                  class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
                >
                  Sed ultricies odio dui, a auctor dolor posuere non. Integer sem mauris, molestie sed auctor vel, dictum eget ligula. Nunc pulvinar ex ut mattis accumsan. Etiam non tortor varius, blandit velit a, blandit lacus.
                </span>
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                public
              </p>
            </div>
          </li>
          <li
            class="MuiDivider-root"
            role="separator"
          />
        </div>
        <div>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Cloud Strife"
                  class="MuiAvatar-img"
                  src="/static/images/avatar/1.jpg"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <span
                  class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
                >
                  Aenean tincidunt mattis risus, pulvinar accumsan ligula semper quis. Class aptent taciti sociosqu ad litora torquent per conubia nostra.
                </span>
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                private
              </p>
            </div>
          </li>
          <li
            class="MuiDivider-root"
            role="separator"
          />
        </div>
        <div>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
          >
            <div
              class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Jin Sakai"
                  class="MuiAvatar-img"
                  src="/static/images/avatar/1.jpg"
                />
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                <span
                  class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
                >
                  Quisque sit amet sollicitudin purus, non venenatis lacus. Aenean a tellus massa. Vestibulum tempor nisi eget lorem suscipit, vel commodo velit feugiat. 
                </span>
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                private
              </p>
            </div>
          </li>
          <li
            class="MuiDivider-root"
            role="separator"
          />
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h5
      class="MuiTypography-root makeStyles-text-3 MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Messages
    </h5>
    <ul
      class="MuiList-root makeStyles-root-1 MuiList-padding"
    >
      <div>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="Lara Croft"
                class="MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <span
                class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                Vivamus hendrerit ipsum eros, non vulputate elit cursus sed. Proin fringilla vestibulum lobortis. Vivamus blandit diam vitae massa iaculis blandit. Sed molestie pharetra nisl, sed lobortis nunc venenatis in. Duis vel leo id elit vestibulum iaculis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent felis augue, aliquam quis nisl et, sollicitudin tincidunt augue.
              </span>
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              public
            </p>
          </div>
        </li>
        <li
          class="MuiDivider-root"
          role="separator"
        />
      </div>
      <div>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="Nathan Drake"
                class="MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <span
                class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                Sed ultricies odio dui, a auctor dolor posuere non. Integer sem mauris, molestie sed auctor vel, dictum eget ligula. Nunc pulvinar ex ut mattis accumsan. Etiam non tortor varius, blandit velit a, blandit lacus.
              </span>
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              public
            </p>
          </div>
        </li>
        <li
          class="MuiDivider-root"
          role="separator"
        />
      </div>
      <div>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="Cloud Strife"
                class="MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <span
                class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                Aenean tincidunt mattis risus, pulvinar accumsan ligula semper quis. Class aptent taciti sociosqu ad litora torquent per conubia nostra.
              </span>
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              private
            </p>
          </div>
        </li>
        <li
          class="MuiDivider-root"
          role="separator"
        />
      </div>
      <div>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="Jin Sakai"
                class="MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
          </div>
          <div
            class="MuiListItemText-root MuiListItemText-multiline"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <span
                class="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
              >
                Quisque sit amet sollicitudin purus, non venenatis lacus. Aenean a tellus massa. Vestibulum tempor nisi eget lorem suscipit, vel commodo velit feugiat. 
              </span>
            </span>
            <p
              class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              private
            </p>
          </div>
        </li>
        <li
          class="MuiDivider-root"
          role="separator"
        />
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
